src/components/Messages/index.tsx

```diff
@@ -1,13 +1,95 @@
+import { Card } from '@components/UI/Card';
 import { GridItemEight, GridItemFour, GridLayout } from '@components/UI/GridLayout';
 import MetaTags from '@components/utils/MetaTags';
+import { Profile } from '@generated/types';
 import isFeatureEnabled from '@lib/isFeatureEnabled';
-import { NextPage } from 'next';
+import { Client, Conversation, Stream } from '@xmtp/xmtp-js';
+import { useRouter } from 'next/router';
+import { FC, useEffect, useState } from 'react';
 import { APP_NAME } from 'src/constants';
 import Custom404 from 'src/pages/404';
 import { useAppStore } from 'src/store/app';
+import { useXmtpStore } from 'src/store/xmtp';
+import { useAccount, useSigner } from 'wagmi';
 
-const Messages: NextPage = () => {
+interface Props {
+  profile: Profile;
+}
+
+const Messages: FC<Props> = () => {
+  const { data: signer } = useSigner();
+  const { address } = useAccount();
   const currentProfile = useAppStore((state) => state.currentProfile);
+  const [stream, setStream] = useState<Stream<Conversation>>();
+  const xmtpState = useXmtpStore((state) => state);
+  const { client, setClient, conversations, setConversations, messages, setMessages, setLoading } = xmtpState;
```

Comment:
**question:** will it select only the used state or affect all the states?

https://github.com/pmndrs/zustand#selecting-multiple-state-slices

Reply:
We can visit this later 🙏🏼